framework/wazuh/core/indexer/models/events.py

```diff
@@ -20,248 +23,227 @@
+class Parent(BaseModel):
     """Process parent data model."""
-    pid: float
+    pid: float = None
 
 
-@dataclass
-class ID:
+class ID(BaseModel):
     """Process users and groups ID data model."""
-    id: str
+    id: str = None
 
 
-@dataclass
-class Process:
+class Process(BaseModel):
     """Process data model."""
-    pid: float
-    name: str
-    parent: Parent
-    command_line: str
-    args: List[str]
-    user: ID
-    real_user: ID
-    saved_user: ID
-    group: ID
-    real_group: ID
-    saved_group: ID
-    start: datetime
-    thread: ID
-
-
-@dataclass
-class InventoryProcessEvent:
+    pid: float = None
+    name: str = None
+    parent: Parent = None
+    command_line: str = None
+    args: List[str] = None
+    user: ID = None
+    real_user: ID = None
+    saved_user: ID = None
+    group: ID = None
+    real_group: ID = None
+    saved_group: ID = None
+    start: datetime = None
+    thread: ID = None
+
+
+class InventoryProcessEvent(BaseModel):
     """Inventory process events data model."""
-    scan_time: datetime
-    process: Process
+    scan_time: datetime = None
+    process: Process = None
 
 
-@dataclass
-class InventorySystemEvent:
+class InventorySystemEvent(BaseModel):
     """Inventory system events data model."""
-    scan_time: datetime
-    host: Host
+    scan_time: datetime = None
+    host: Host = None
 
 
-@dataclass
-class SCAEvent:
+class SCAEvent(BaseModel):
     """SCA events data model."""
     # TODO(25121): Add SCA event fields once they are defined
 
 
-@dataclass
-class VulnerabilityEventHost:
+class VulnerabilityEventHost(BaseModel):
     """Host data model in relation to vulnerability events."""
-    os: OS
+    os: OS = None
 
 
-@dataclass
-class VulnerabilityEventPackage:
+class VulnerabilityEventPackage(BaseModel):
     """Package data model in relation to vulnerability events."""
-    architecture: str
-    build_version: str
-    checksum: str
-    description: str
-    install_scope: str
-    installed: datetime
-    license: str
-    name: str
-    path: str
-    reference: str
-    size: float
-    type: str
-    version: str
+    architecture: str = None
+    build_version: str = None
+    checksum: str = None
+    description: str = None
+    install_scope: str = None
+    installed: datetime = None
+    license: str = None
+    name: str = None
+    path: str = None
+    reference: str = None
+    size: float = None
+    type: str = None
+    version: str = None
 
 
-@dataclass
-class Cluster:
+class Cluster(BaseModel):
     """Wazuh cluster data model."""
-    name: str
-    node: str
+    name: str = None
+    node: str = None
 
 
-@dataclass
-class Schema:
+class Schema(BaseModel):
     """Wazuh schema data model."""
-    version: str
+    version: str = None
 
 
-@dataclass
-class Wazuh:
+class Wazuh(BaseModel):
     """Wazuh instance information data model."""
-    cluster: Cluster
-    schema: Schema
+    cluster: Cluster = None
+    schema: Schema = None
```

Comment:
I'm seeing this warning at the comms API start up

```console
/var/lib/wazuh-server/framework/python/lib/python3.10/site-packages/pydantic/_internal/_fields.py:172: UserWarning: Field name "schema" in "Wazuh" shadows an attribute in parent "BaseModel"
  warnings.warn(
```
Could we rename this attribute?

Reply:
This is going to be done in https://github.com/wazuh/wazuh/issues/26656 because the model is outdated